shell/platform/android/io/flutter/embedding/engine/dart/DartExecutor.java

```diff
@@ -181,6 +184,22 @@ public void setMessageHandler(@NonNull String channel, @Nullable BinaryMessenger
   }
   //------ END BinaryMessenger -----
 
+  /**
+   * Returns an identifier for this executor's primary isolate.  This identifier can be used
+   * in queries to the Dart service protocol.
+   */
+  public String getIsolateServiceId() {
+    return isolateServiceId;
+  }
+
+  private BinaryMessenger.BinaryMessageHandler isolateChannelMessageHandler =
+      new BinaryMessenger.BinaryMessageHandler() {
+        @Override
+        public void onMessage(ByteBuffer message, final BinaryReply callback) {
+          isolateServiceId = StringCodec.INSTANCE.decodeMessage(message);
```

Comment:
I'm wondering if we need a listener for this? The value is not available immediately upon construction, but instead appears an unknown time later. I'm not sure how clients will know, in general, when they can query this...but I can always add that later if we're not sure if we need it.

Reply:
Added a listener interface